src/lando/api/legacy/workers/landing_worker.py

```diff
@@ -234,131 +235,135 @@ def run_job(self, job: LandingJob) -> bool:
             )
             return False
 
-        with scm.for_push(job.requester_email):
-            # Update local repo.
-            repo_pull_info = f"tree: {repo.tree}, pull path: {repo.pull_path}"
-            try:
-                scm.update_repo(repo.pull_path, target_cset=job.target_commit_hash)
-            except SCMInternalServerError as e:
-                message = (
-                    f"`Temporary error ({e.__class__}) "
-                    f"encountered while pulling from {repo_pull_info}"
-                )
-                logger.exception(message)
-                job.transition_status(LandingJobAction.DEFER, message=message)
-
-                # Try again, this is a temporary failure.
-                return False
-            except Exception as e:
-                message = f"Unexpected error while fetching repo from {repo.name}."
-                logger.exception(message)
-                job.transition_status(
-                    LandingJobAction.FAIL,
-                    message=message + f"\n{e}",
-                )
-                self.notify_user_of_landing_failure(job)
-                return True
-
-            # Run through the patches one by one and try to apply them.
-            for revision in job.revisions.all():
-
+        with PushLogForRepo(repo, job.requester_email) as pushlog:
```

Comment:
This diff is better viewed with `Hide whitespace` enabled.

![2025-01-30T16:30:12,436737036+11:00](https://github.com/user-attachments/assets/fda8092e-1c1a-4e27-a216-df2b8c5bd5f9)